torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -7339,19 +7339,38 @@ def forward(self, x):
         @require_backends_available(DistTestCases.backend_feature["gpu"])
         @skip_if_lt_x_gpu(2)
         def test_different_graph_across_ranks(self):
+            # In default rebuilt bucket is enabled when
+            # find_unused_parameters=True.
             base_model = self._test_different_graph_across_ranks(
                 find_unused_parameters=True
             )
-            self.assertFalse(
+            self.assertTrue(
                 base_model._get_ddp_logging_data().get("has_rebuilt_buckets", 0)
             )
+
+            # rebuilt bucket could be disabled when find_unused_parameters=True
+            # by setting environment variable for debugging purpose.
+            os.environ["DISABLE_REBUILT_BUCKET"] = "1"
```

Comment:
nit: DISABLE_REBUILD_BUCKET instead of DISABLE_REBUILT_BUCKET

Also the main comment - IMO this can easily get a bit confusing, DISABLE_REBUILD_BUCKET seems to imply it will _always_ be disabled no matter the DDP settings, but it actually is contingent on find_unused_parameters=True. 

Do we absolutely need this flag or is it there for preemptive purposes such as helping if something breaks down the line?